src/content/docs/tutorials/crates-tui/app.md

```diff
@@ -0,0 +1,106 @@
+---
+title: App
+---
+
+We saw in the previous sections that `main` looked like this:
+
+```rust
+{{#include @code/crates-tui-tutorial-app/src/main.rs:main}}
+```
+
+In this section, we are going to define our `App` struct.
+
+Our app is going to have two focus modes:
+
+1. when the `Prompt` is in focus,
+2. when the `Results` are in focus.
+
+And we can represent this using an enum `Mode`:
+
+```rust
+{{#include @code/crates-tui-tutorial-app/src/app.rs:mode}}
+```
+
+Let's define an `App` struct with the following fields:
+
+```rust
+{{#include @code/crates-tui-tutorial-app/src/app.rs:app}}
+```
+
+One of the first steps to building truly `async` TUI applications is to use the `Command`, `Action`,
+or `Message` pattern.
+
+:::tip
+
+The `Command` pattern is the concept of "reified method calls". You can learn a lot more about this
+pattern from the excellent
+[http://gameprogrammingpatterns.com](http://gameprogrammingpatterns.com/command.html).
```

Comment:
Reified isn't a commonly used term, using it to describe how things work in our docs makes things more complex rather than simpler.

Reply:
I removed the term and used a modified definition from refactoring guru.